### DISCUSSION TITLE
Is there any sample available for Sankey chart ?

### DISCUSSION THREAD

**Discussion by gokulks** [jfree/jfreechart#341] [Q&A] [answered]
Hi All,

I couldn't find any example for the Sankey chart over the internet. Is there any chance of getting examples / demos for the same. 

Thanks in advance .

**Comment by trashgod** [comment, marked as answer]
[`org.jfree.chart.plot.flow.FlowPlot`](https://www.jfree.org/jfreechart/javadoc/org/jfree/chart/plot/flow/FlowPlot.html) is "a type of Sankey chart." The original project is now [archived](https://github.com/jfree/flowplot), but it may yield some insights.

**Reply by gokulks** [reply]
Thanks for your reply. Am able to construct the chart.
If this is my data flow say, "Salary"->   "Earned Income"-> "Income" ->  "Deduction"-> "Income Tax" and below is the JSON data
 {
        "Income Source": "Salary",
        "Income Type": "Earned Income",
        "Income": "Income",
        "Spending Source": "Deduction",
        "Spending Type": "Income Tax",
        "Amount": 5040
    }
1 ) where can we set the value(Amount)?
2) the last destination label **"Income Tax**" is aligned left unlike others, is it possible to align Right

![image](https://user-images.githubusercontent.com/1009845/216135424-36850e74-bc88-4b78-b3d0-f015e0573994.png)
3) Is it possible to duplicate the flow if there is a common source.
![image](https://user-images.githubusercontent.com/1009845/216137848-64c8bd0c-a008-4828-beee-990d57370108.png)
Here, Salary-Social Justice and Salary -Bill exp  we have common sources, but there 2 separate flows showing all the stages with different shades. Is this possible with Jfree?

**Comment by trashgod** [comment]
I'm not sure, but I see that `FlowPlotDemo1` has a `FlowDataset` with three stages that might suggest a way forward.
